cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -555,6 +555,7 @@ private void startExecution(LoadMonitor loadMonitor,
     sanityCheckOngoingMovement();
     _hasOngoingExecution = true;
     _anomalyDetector.maybeClearOngoingAnomalyDetectionTimeMs();
+    _anomalyDetector.resetHasUnfixableGoals();
```

Comment:
Why we need to reset `hasUnfixableGoals` here? Shouldn't refreshing this value in anomaly detector's anomaly handling logic enough?

Reply:
An ongoing execution would potentially take hours to days to complete. During an ongoing execution, goal violation detector skips detection; hence, would fail to refresh this state.

Resetting the state during an ongoing execution avoids keeping this signal for non-actionable case. If an execution was manually started, the cluster operator already knows if there are unfixable goals, because s/he either fixes them or skips them explicitly (on purpose).

Reply:
I see, thanks for the explanation.